### ISSUE TITLE
Ensure handling all errors

### ISSUE THREAD

**Issue by steida** [pigoz/effect-crashcourse#12]
It would be nice to add an example demonstrating handling all errors with an exhaustive switch (or is Match better?)

**Comment by pigoz** [comment]
Right, I'm actually not sure what the prescribed way is. In my app I have a function that takes in `Effect<R,never,A>` and passes it to the Runtime<R>.

You can do exaustive switchs with match, but I believe something leveraging catchTags would be better? i.e.:

```typescript
pipe(
  Effect.fail({ _tag: "foo" as const }),
  Effect.catchTag("foo", (_) => Effect.die(_)),
  Effect.mapError<never, never>(identity) // <- I wonder if this should be called Effect.exaustive
);
```

**Comment by pigoz** [comment]
cc @mikearnaldi

**Comment by steida** [comment]
@pigoz Maybe match?

```ts
const p = pipe(
  sync(req),
  Effect.provideService(DbTag, db),
  Effect.match(
    (e) => {
      //
    },
    (a) => {
      //
    }
  )
);
```

Maybe `runCallback` is better, but then how to handle Exit with all known and unknown errors.

**Comment by pigoz** [comment]
I thinks it's something that depends on your app. Like, if your Effects are ultimately converted into Promises, you can make a `runPromiseExaustive` function:

```typescript
function runPromiseExaustive<A>(effect: Effect.Effect<never, never, A>) {
  return Effect.runPromise(effect);
}

function runtimeRunPromiseExaustive(runtime: Runtime.Runtime<R>, effect: Effect.Effect<R, never, A>) {
  return runtime.runPromise(effect)
}
```

That's the approach I'm using currently, since I'm using Effect in the server side parts of Remix. Also if you are splitting Effect creations inside functions you can specify E as never as the return type.

But a way to assert there's no unhandled failures in a pipe might be useful.
